tests/queries/0_stateless/02942_variant_cast.sql

```diff
@@ -1,5 +1,4 @@
 set allow_experimental_variant_type=1;
-set allow_experimental_analyzer=0; -- It's currently doesn't work with analyzer because of the way it works with constants, but it will be refactored and fixed in future
```

Comment:
I didn't know that there were any issues with Variant data type. @Avogar Could you please comment of this?

Reply:
https://github.com/ClickHouse/ClickHouse/pull/58047#discussion_r1452530196

Should be fixed after constants refacroting in Analyzer when we get rid of using `Field`.

Reply:
It's better to remove `set allow_experimental_analyzer = 0` and add test to the `tests/analyzer_tech_debt.txt`. Otherwise, we'll forget about these tests.